docs/graphcast/radios/graphcast-cli.md

```diff
@@ -0,0 +1,95 @@
+---
+sidebar_position: 3
+---
+
+# Graphcast CLI
+
+The source code for the Graphcast CLI is available [on GitHub](https://github.com/graphops/graphcast-cli).
+
+## Introduction
+
+The Graphcast CLI enables sending one-off messages. Currently, its default behaviour is to construct and send a message of type `UpgradeIntentMessage`, which is used for the [Subgraph Upgrade Pre-sync feature](https://docs.graphops.xyz/graphcast/design-principles#subgraph-upgrade-pre-sync) of Subgraph Radio.
+
+The Graphcast CLI is configured using config variables. You will need to prepare the following config variables (either as env variables or passing CLI args when running the CLI):
+
+| Name                | Description and Examples                                                                                                                                        |
+| ------------------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------- | --- |
+| `PRIVATE_KEY`       | Private key to the Graphcast ID wallet (precendence over mnemonics).<br/>Example: `PRIVATE_KEY=YOUR_PRIVATE_KEY`                                                |
+| `MNEMONIC`          | Mnemonic to the Graphcast ID wallet (first address of the wallet is used; Only one of private key or mnemonic is needed).<br/>Example: `MNEMONIC=YOUR_MNEMONIC` |
+| `GRAPH_ACCOUNT`     | Graph account corresponding to Graphcast operator.<br/>Example: `GRAPH_ACCOUNT=YOUR_GRAPH_ACCOUNT`                                                              |
+| `REGISTRY_SUBGRAPH` | Subgraph endpoint to the Graphcast Registry.<br/>Default: `https://api.thegraph.com/subgraphs/name/hopeyen/graphcast-registry-goerli`                           |
+| `NETWORK_SUBGRAPH`  | Subgraph endpoint to The Graph network subgraph.<br/>Default: `https://api.thegraph.com/subgraphs/name/graphprotocol/graph-network-goerli`                      |
+| `GRAPHCAST_NETWORK` | Supported Graphcast networks: mainnet, testnet.<br/>Default: `testnet`                                                                                          |     |
+| `LOG_LEVEL`         | Logging configuration to set as RUST_LOG.<br/>Default: `info`                                                                                                   |
+| `LOG_FORMAT`        | Support logging formats: pretty, json, full, compact.<br/>Default: `pretty`                                                                                     |
+
+The Graphcast CLI code is very extensible and could be altered to send any kind of Graphcast-compatible message to the network.
+
+## Run with Docker
```

Comment:
Perhaps include a paragraph on what to expect when they run this command (potential logs and summaries)

Reply:
maybe we can have a more dedicated page for the presyncing feature overall, with concise description on the ideal flow and interaction of components. like the usage guide you mentioned writing